File: StateDemo/StateDemo/MyClass.swift
//
//  MyClass.swift
//  StateDemo
//
//  Created by Andre Pham on 31/3/2023.
//

import Foundation
import SwiftUI

class MyClass: ObservableObject {
    
    @Published var text: String
    
    init(text: String) {
        self.text = text
    }
    
}

File: LearningState/MyClass.ts

class MyClass {

    private text: string;

    constructor(text: string) {
        this.text = text
    }

    public setText(text: string) {
        this.text = text
        console.log("this.text = " + this.text)
    }

    public getText(): string {
        return this.text
    }

}

export default MyClass